src/QMCWaveFunctions/lcao/SoaAtomicBasisSet.h

```diff
@@ -27,15 +27,15 @@ namespace qmcplusplus
     struct SoaAtomicBasisSet
     {
       typedef ROT RadialOrbital_t;
-      typedef typename ROT::value_type value_type;
+      using ValueType = typename ROT::value_type;
```

Comment:
ValueType in the source code is reserved for data types turning complex when the code is built complex.
In the AO basis, it is always real only even in the future when complex support of LCAO is added.
Please restore this change as well as all the ValueType

Reply:
ValueType having a special global meaning like that is a design issue. You should be able to build complex and real simultaneously and the local meaning of a type should not require special global knowledge of the code. This is the ValueType for a radial function used by this basis set. It must be the type returned by RF(r).  This could be mathematically a real, complex, quaternion or whatever.

Reply:
I think the reason of using value_type is to do some local template. ValueType is reserved for global meeting. This is an convention. The real issue is mixing them here.